src/secondary_threads/parse_packets.rs

```diff
@@ -33,6 +33,11 @@ pub fn parse_packets(
 ) {
     let capture_id = *current_capture_id.lock().unwrap();
 
+    // pcap seems to assign ethernet to all interfaces (at least on macOS)...
```

Comment:
I see a different behavior on my MacOS. If I run

```rust
fn main() {
    // list all of the devices pcap tells us are available
    for device in pcap::Device::list().expect("device lookup failed") {
        println!("Found device! {:?}", device);
        
        let Ok(cap) = pcap::Capture::from_device(device) else {
            println!("    Error creating capture");
            continue;
        };
        let Ok(cap) = cap.immediate_mode(true).open() else {
            println!("    Error opening capture");
            continue;
        };

        {
            let l = cap.get_datalink();
            println!("  Main data link: {:?} {:?} {:?}", l, l.get_name(), l.get_description())
        }

        if let Ok(links) = cap.list_datalinks() {
            for l in links {
                println!("    {:?} {:?} {:?}", l, l.get_name(), l.get_description());
            }
        }
    }
}
```

I get my loopback device correctly reported as "LOOP":

```
...
Found device! Device { name: "lo0", desc: None, addresses: [Address { addr: 127.0.0.1, netmask: Some(255.0.0.0), broadcast_addr: None, dst_addr: None }, Address { addr: ::1, netmask: Some(ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff), broadcast_addr: None, dst_addr: None }, Address { addr: fe80::1, netmask: Some(ffff:ffff:ffff:ffff::), broadcast_addr: None, dst_addr: None }], flags: DeviceFlags { if_flags: LOOPBACK | UP | RUNNING, connection_status: NotApplicable } }
  Main data link: Linktype(0) Ok("NULL") Ok("BSD loopback")
    Linktype(0) Ok("NULL") Ok("BSD loopback")
...
```

Maybe recheck on your system again, then you could potentially remove some logic.

Reply:
Solved this one.
The problem was that I was creating the capture from the device name (`&str`) instead of opening it from the device itself (`pcap::device`).